File: src/leetcode/julia/arrays_hashing/LongestConsecutive.jl
function longestConsecutive(nums::Vector{Int})::Int
    if isempty(nums)
        return 0
    end

    nums_set = Set(nums)
    longest = 0

    for num in nums_set
        # Check if it's the start of a sequence
        if !((num - 1) in nums_set)
            current_num = num
            current_length = 1
            # Count the length of the current consecutive sequence
            while (current_num + 1) in nums_set
                current_num += 1
                current_length += 1
            end
            longest = max(longest, current_length)
        end
    end
    return longest
end

using Test

@testset "Longest Consecutive Sequence Tests" begin
    @test longestConsecutive([100, 4, 200, 1, 3, 2]) == 4
    @test longestConsecutive([0, 3, 7, 2, 5, 8, 4, 6, 0, 1]) == 9
    @test longestConsecutive(Int[]) == 0 # Empty array
    @test longestConsecutive([1]) == 1 # Single element
    @test longestConsecutive([1, 1, 1, 1]) == 1 # Duplicate elements, single sequence
    @test longestConsecutive([1, 2, 3, 5, 6]) == 3 # Two sequences
    @test longestConsecutive([5, 4, 3, 2, 1]) == 5 # Reverse order
    @test longestConsecutive([1, 3, 5, 7, 9]) == 1 # No consecutive elements
    @test longestConsecutive([9, 1, 4, 7, 3, -1, 0, 5, 8, -1, 6]) == 7 # More complex case from LeetCode
end

File: src/leetcode/python/arrays_hashing/longest_consecutive.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        # convert it to an unordered set with unique elements
        nums_set = set(nums)
        counter = 0

        for num in nums_set:
            if num - 1 not in nums_set:
                length = 1

                while (num + length) in nums_set:
                    length += 1
                # check if we need to refresh the longest counter after tabulating the consecutive length
                counter = max(counter, length)

        return counter


if __name__ == "__main__":
    solver = Solution()
    assert solver.longestConsecutive([2, 20, 4, 10, 3, 4, 5]) == 4
    assert solver.longestConsecutive([0, 3, 2, 5, 4, 6, 1, 1]) == 7
